apps/search/js/providers/suggestions.js

```diff
@@ -32,6 +36,10 @@
     suggestionsSelect.appendChild(selectFragment);
   });
 
+  SettingsListener.observe(enabledKey, true, function(enabled) {
```

Comment:
Should we just access the state from the global Search object? It looks like we're already observing that setting there..

https://github.com/mozilla-b2g/gaia/blob/master/apps/search/js/search.js#L84